third_party/chromium/media/filters/source_buffer_stream.cc

```diff
@@ -1900,11 +1900,14 @@ bool SourceBufferStream::UpdateVideoConfig(const VideoDecoderConfig& config,
   video_configs_[append_config_index_] = config;
 
 #if defined(STARBOARD)
-  // Dynamically increase |memory_limit_| when video resolution goes up.
-  memory_limit_ = std::max(
-      memory_limit_,
-      GetDemuxerStreamVideoMemoryLimit(Demuxer::DemuxerTypes::kChunkDemuxer,
-                                       &config, mime_type_));
+  // Dynamically increase |memory_limit_| when video resolution goes up as long
+  // as we haven't set a manual override.
+  if (!memory_override_) {
+    memory_limit_ = std::max(
```

Comment:
Sorry that I cannot comment on line 819.  If |memory_limit_| is set in the ctor as is, then line 819 won't be able to pick up the value set in SourceBuffer.
My brief thought would be to replace functions like GetDemuxerStreamAudioMemoryLimit() by functors, and inject them from ChunkDemuxer or WebMediaPlayerImpl or both, but we probably need to think through a detailed plan.